spec/units/commands/git_command_spec.rb

```diff
@@ -2,43 +2,34 @@
 require 'gitsh/commands/git_command'
 
 describe Gitsh::Commands::GitCommand do
-  let(:env) do
-    double('Environment', {
-      git_command: '/usr/bin/env git',
-      output_stream: double('OutputStream', to_i: 1),
-      error_stream: double('ErrorStream', to_i: 2)
-    })
-  end
-
-  before do
-    allow(Process).to receive(:spawn).and_return(1)
-    allow(Process).to receive(:wait)
-  end
-
   describe '#execute' do
-    it 'spawns a process with the sub command and arguments' do
-      allow(env).to receive(:config_variables).and_return({})
-      command = described_class.new(
-        env,
-        'commit',
-        arguments('-m', 'A test commit'),
-      )
+    it 'delegates to the Gitsh::ShellCommandRunner' do
+      env = double(:env, git_command: '/usr/bin/env git', config_variables: {})
+      expected_result = double(:result)
+      mock_runner = double(:shell_command_runner, run: expected_result)
+      allow(Gitsh::ShellCommandRunner).to receive(:new).and_return(mock_runner)
 
-      command.execute
+      command = described_class.new(env, "commit", arguments("-m", "Some stuff"))
+      result = command.execute
 
-      expect(Process).to have_received(:spawn).with(
-        '/usr/bin/env', 'git',
-        'commit',
-        '-m', 'A test commit',
-        out: env.output_stream.to_i,
-        err: env.error_stream.to_i
+      expect(Gitsh::ShellCommandRunner).to have_received(:new).with(
```

Comment:
The strategy in this project is to pass all classes as arguments to the constructor, and to pass a double in the test. Here, you do it differently. Any reason?

(IIRC, `Process.spawn` was mocked here because it was tricky to do it otherwise.)